src/components/Header/Header.js

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react';
+
+const Header = (props) => {
+  const [todo, setTodo] = useState('');
+
+  const onKeyPressHandler = (event) => {
+    if(event.key === 'Enter' && todo){
+      const todoItem = {
+        id: new Date().valueOf(),
+        value: todo,
+        isCompleted: false
+      };
+      props.changed(todoItem);
+      setTodo('');
+    }
+  };
+
+  const onChangeTextHandler = (event) => {
+    const currentValue = event.target.value;
+    if(currentValue.trim() !== '') {
+      setTodo(event.target.value);
+    }
+  }
+
+  return( <header className="header">
+      <h1>todos</h1>
+      <input
+        className="new-todo"
+        placeholder="What needs to be done?"
+        onChange={(event => onChangeTextHandler(event))}
+        onKeyPress={(event) => onKeyPressHandler(event)}
```

Comment:
why don't you define one function for both `onChange` and `onKeyPress`

Reply:
I need catch onChange event to manage the state => update value of <input/> element. I do need onKeyChange to define the way use put their work into to-do list. If i only use onChange the list will take an unnecessary checking to update. On the other hand, if i only use onKeyPress, the state can not update immediately and user can not see what their entered.